verde/coordinates.py

```diff
@@ -702,3 +728,99 @@ def block_split(coordinates, spacing, adjust="spacing", region=None):
     tree = kdtree(block_coords)
     labels = tree.query(np.transpose(n_1d_arrays(coordinates, 2)))[1]
     return block_coords, labels
+
+
+def latlon_continuity(region, coordinates=None):
+    """
+    Modify geographic region boundaries to ensure continuity around the globe.
+
+    Longitudinal boundaries of the region are moved to the `[0, 360)` or `[-180, 180)`
+    degrees interval depending which one is better suited for that specific region.
+
+    Parameters
+    ----------
+    region : list or array
+        List or array containing the boundary coordinates `w, `e`, `s`, `n` of the
+        region in degrees.
+    coordinates : list or array (optional)
+        Extra set of geographic coordinates that will be moved to the same degrees
+        interval as the one of the modified region.
+
+    Returns
+    -------
+    modified_region : array
+        List containing the modified boundary coordinates `w, `e`, `s`, `n` of the
+        region.
+    modified_coordinates : array (optional)
+        Modified set of extra geographic coordinates.
+
+    Examples
+    --------
+
+    >>> from verde import latlon_continuity
+    >>> # Modify region with west > east
+    >>> w, e, s, n = 350, 10, -10, 10
+    >>> print(latlon_continuity([w, e, s, n]))
+    [-10  10 -10  10]
+    >>> # Modify region and extra coordinates
+    >>> from verde import grid_coordinates
+    >>> region = [-70, -60, -40, -30]
+    >>> coordinates = grid_coordinates([270, 320, -50, -20], spacing=5)
+    >>> region, [longitude, latitude] = latlon_continuity(region, coordinates)
+    >>> print(region)
+    [290 300 -40 -30]
+    >>> print(longitude.min(), longitude.max())
+    270.0 320.0
+    >>> # Another example
+    >>> region = [-20, 20, -20, 20]
+    >>> coordinates = grid_coordinates([0, 350, -90, 90], spacing=10)
+    >>> region, [longitude, latitude] = latlon_continuity(region, coordinates)
+    >>> print(region)
+    [-20  20 -20  20]
+    >>> print(longitude.min(), longitude.max())
+    -180.0 170.0
+    """
+    # Get longitudinal boundaries and check region
+    w, e, s, n = region[:4]
+    check_region([w, e, s, n], latlon=True)
+    # Check if region is defined all around the globe
+    all_globe = np.allclose(abs(e - w), 360)
+    # Move coordinates to [0, 360)
+    interval_360 = True
+    w = w % 360
+    e = e % 360
+    # Move west=0 and east=360 if region longitudes goes all around the globe
+    if all_globe:
+        w, e = 0, 360
```

Comment:
I've added this `all_globe` check in case you pass `w, e = 0, 360` as region boundaries.
If we omit the `all_globe`, the algorithm will return `w, e = 0, 0`, which is problematic.
The *slightly off coordinates** you mention will be within an error of 1e-5 (according to `numpy.allclose`).
One simple hack would be to lower down this error, but I don't know if it's ideal.

What do you think?

Reply:
My thinking for this is "do we need to do this check?". If this is not necessary, then we could skip the check and avoid the issues with precision.